Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp

```diff
@@ -2685,31 +2685,41 @@ Ref<DownloadProxy> WebsiteDataStore::createDownloadProxy(Ref<API::DownloadClient
     return protectedNetworkProcess()->createDownloadProxy(*this, WTFMove(client), request, frameInfo, originatingPage);
 }
 
+void WebsiteDataStore::prepareRequestForDownload(WebCore::ResourceRequest& request, WebPageProxy* originatingPage)
+{
+    // Request's firstPartyForCookies will be used as Original URL of the download request.
+    // We set the value to top level document's URL.
+    if (originatingPage) {
+        URL originatingPageURL = URL { originatingPage->pageLoadState().url() };
+        request.setFirstPartyForCookies(originatingPageURL);
+        request.setIsSameSite(WebCore::areRegistrableDomainsEqual(originatingPageURL, request.url()));
+        if (!request.hasHTTPHeaderField(WebCore::HTTPHeaderName::UserAgent))
+        request.setHTTPUserAgent(originatingPage->userAgentForURL(request.url()));
+    } else {
+        request.setFirstPartyForCookies(URL());
+        request.setIsSameSite(false);
+        if (!request.hasHTTPHeaderField(WebCore::HTTPHeaderName::UserAgent))
+        request.setHTTPUserAgent(WebPageProxy::standardUserAgent());
```

Comment:
```suggestion
        if (!request.hasHTTPHeaderField(WebCore::HTTPHeaderName::UserAgent))
            request.setHTTPUserAgent(WebPageProxy::standardUserAgent());
```

Reply:
beyond the mere formatting suggestions above, not sure if it's cleaner to move the user agent logic outside, something like

```
if (!request.hasHTTPHeaderField(WebCore::HTTPHeaderName::UserAgent))
    request.setHTTPUserAgent(originatingPage ? originatingPage->userAgentForURL(request.url()) : WebPageProxy::standardUserAgent());
```

Reply:
No strong preference. Was just keeping the logic unchanged.